crates/sp-domains/src/lib.rs

```diff
@@ -1628,12 +1628,52 @@ impl SkipBalanceChecks for () {
     }
 }
 
+/// Represents a nominator's storage fee deposit information
+#[derive(Debug, Encode, Decode, TypeInfo, Clone, PartialEq, Eq)]
+pub struct StorageFeeDeposit<Balance> {
+    /// Original amount contributed to storage fees
+    pub total_deposited: Balance,
+    /// Current value adjusted for fund performance (gains/losses)
+    pub current_value: Balance,
+}
+
+/// Represents a nominator's pending deposit that hasn't been converted to shares yet
+#[derive(Debug, Encode, Decode, TypeInfo, Clone, PartialEq, Eq)]
+pub struct PendingDeposit<Balance> {
+    /// The amount of the pending deposit
+    pub amount: Balance,
+    /// The epoch when this deposit will become effective
+    pub effective_epoch: EpochIndex,
+}
+
+/// Represents a nominator's pending withdrawal with unlock timing
+#[derive(Debug, Encode, Decode, TypeInfo, Clone, PartialEq, Eq)]
+pub struct PendingWithdrawal<Balance, DomainBlockNumber> {
+    /// The amount that will be withdrawn
+    pub amount: Balance,
+    /// The domain block number when this withdrawal can be unlocked
+    pub unlock_at_block: DomainBlockNumber,
+}
+
+/// Complete nominator position information for a specific operator
+#[derive(Debug, Encode, Decode, TypeInfo, Clone, PartialEq, Eq)]
+pub struct NominatorPosition<Balance, DomainBlockNumber> {
+    /// Current value of the nominator's position (stake converted to balance using current share price)
+    pub current_staked_value: Balance,
+    /// Storage fee deposit information (original and current adjusted values)
+    pub storage_fee_deposit: StorageFeeDeposit<Balance>,
+    /// Pending deposits not yet converted to shares
+    pub pending_deposits: Vec<PendingDeposit<Balance>>,
```

Comment:
There is at most one pending deposit thus `pending_deposits: Option<PendingDeposit<Balance>>` may be more appropriate.